scipy/optimize/tests/test_zeros.py

```diff
@@ -56,6 +56,44 @@ def test_newton(self):
             x = zeros.newton(f, 3, fprime=f_1, fprime2=f_2, tol=1e-6)
             assert_allclose(f(x), 0, atol=1e-6)
 
+    def test_newton_array(self):
+        """test newton with array"""
+
+        def f_solarcell(i, v, il, io, rs, rsh, vt):
```

Comment:
Better to avoid domain specific function names and variables--in the far future nobody will remember what they're from.

Reply:
✔️